Makefile

```diff
@@ -0,0 +1,41 @@
+# These targets are not files
+.PHONY: lint lint-scripts check-format format run publish
+
+STACK ?= heroku-24
+FIXTURE ?= test/spec/fixtures/repos/java-overlay-test
+
+# Converts a stack name of `heroku-NN` to its build Docker image tag of `heroku/heroku:NN-build`.
+STACK_IMAGE_TAG := heroku/$(subst -,:,$(STACK))-build
+
+lint: lint-scripts check-format
+
+lint-scripts:
+	@git ls-files -z --cached --others --exclude-standard 'bin/*' 'etc/*' 'lib/*' 'opt/*' | xargs -0 shellcheck --check-sourced --color=always
+
+check-format:
+	@shfmt -f . | grep -v "vendor/" | grep -v "test/spec/fixtures/" | xargs shfmt -i 2 --diff
```

Comment:
(optional) An alternative way to ignore certain repo content is to use `ignore = true` in an `.editorconfig`, like in:
https://github.com/heroku/heroku-buildpack-python/blob/main/.editorconfig